casper/src/main/scala/coop/rchain/casper/util/comm/DeployRuntime.scala

```diff
@@ -62,39 +67,38 @@ object DeployRuntime {
       }.map(kp("")).value
     }
 
-  //Accepts a Rholang source file and deploys it to Casper
+//Accepts a Rholang source file and deploys it to Casper
   def deployFileProgram[F[_]: Monad: Sync: DeployService](
       purseAddress: String,
       phloLimit: Long,
       phloPrice: Long,
       nonce: Int,
       validAfterBlock: Int,
-      maybeUserId: Option[PublicKey],
+      maybePrivateKey: Option[PrivateKey],
       file: String
   ): F[Unit] =
     gracefulExit(
       Sync[F].delay(Try(Source.fromFile(file).mkString).toEither).flatMap {
         case Left(ex) =>
           Sync[F].delay(Left(Seq(s"Error with given file: \n${ex.getMessage}")))
         case Right(code) =>
-          val userId =
-            maybeUserId
-              .map(uid => ByteString.copyFrom(uid.bytes))
-              .getOrElse(ByteString.EMPTY)
-
           for {
             timestamp <- Sync[F].delay(System.currentTimeMillis())
+
             //TODO: allow user to specify their public key
             d = DeployData()
               .withTimestamp(timestamp)
               .withTerm(code)
               .withFrom(purseAddress)
               .withPhloLimit(phloLimit)
               .withPhloPrice(phloPrice)
-              .withUser(userId)
               .withNonce(nonce)
               .withValidAfterBlockNumber(validAfterBlock)
-            response <- DeployService[F].deploy(d)
+              .withTimestamp(timestamp)
+            
+            signedData = maybePrivateKey.fold(d)(SignDeployment(_, d))
```

Comment:
`SingDeployment.apply` returning a DeployData is quite a misleading API. I would expect that apply to return a `SignDeployment`. I'd prefer `DeploySigner.sign(key, d)`. Not a big deal though.